### PR TITLE
test(integ): add integ tests for build & publish

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
Adding integration tests for build and publish commands with 90% coverage. 

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [x] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [x] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: integration_tests/gdk/components/test_integ_BuildCommand.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+def test_build_run_default_maven_yaml_error(mocker, supported_build_system, rglob_build_file):
+
+    mock_clean_dir = mocker.patch("gdk.common.utils.clean_dir", return_value=None)
+    mock_create_dir = mocker.patch("pathlib.Path.mkdir", return_value=None)
+    mock_copy_dir = mocker.patch("shutil.copytree", return_value=None)
+    mock_archive_dir = mocker.patch("shutil.make_archive", return_value=None)
+    mock_platform = mocker.patch("platform.system", return_value="Windows")
+    pc = project_config()
+    pc["component_build_config"] = {"build_system": "maven"}
+    pc["component_recipe_file"] = Path("/src/GDK-CLI-Internal/tests/gdk/static/build_command/recipe.yaml")
```
Comment (avipinku) [inline integration_tests/gdk/components/test_integ_BuildCommand.py:238]: nit: can be a constant, especially if you're reusing any portion of this URI

File: integration_tests/gdk/components/test_integ_BuildCommand.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+    mock_subprocess_run = mocker.patch("subprocess.run")
+    mock_yaml_dump = mocker.patch("yaml.dump")
+    pc = mock_get_proj_config.return_value
+
+    with patch("builtins.open", mock_open()) as mock_file:
+        with pytest.raises(Exception) as e:
+            parse_args_actions.run_command(CLIParser.cli_parser.parse_args(["component", "build"]))
+            assert (
+                "Could not find artifact with URI"
+                " 's3://DOC-EXAMPLE-BUCKET/artifacts/com.example.HelloWorld/1.0.0/hello_world.py' on s3 or inside"
```
Comment (avipinku) [inline integration_tests/gdk/components/test_integ_BuildCommand.py:287]: nit: can reuse URI

File: integration_tests/gdk/components/test_integ_BuildCommand.py [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+    assert mock_copy_dir.call_count == 0  # No copying directories
+    assert supported_build_system.call_count == 1
+    assert mock_is_artifact_in_build.call_count == 1
+    assert mock_archive_dir.call_count == 0  # Archvie never called in gralde
+    assert mock_boto3_client.call_count == 0  # artifacts found in s3
+    assert mock_clean_dir.call_count == 1  # clean greengrass-build
+    assert mock_create_dir.call_count == 2  # create gg directories
+
+
+def project_config():
```
Comment (avipinku) [inline integration_tests/gdk/components/test_integ_BuildCommand.py:463]: can this be put into testData as a config file and loaded in from there instead?

**Review by avipinku — DISMISSED** [review]

**Review by pallavi03 — DISMISSED** [review]

**Review by avipinku — APPROVED** [review]

**Review by popanmol — APPROVED** [review]

File: integration_tests/gdk/components/test_integ_PublishCommand.py [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+            "An error occurred (BucketAlreadyExists) when calling the CreateBucket operation: fake message" in e.value.args[0]
+        )
+    assert mock_build_dir_exists.call_count == 1  # Checks if build directory exists
+    assert mock_iter_dir.call_count == 1  # Checks if there is at least one artifact to upload
+    assert mock_glob.call_count == 0  # Recipe is not updated
+
+    # Assert cloud calls
+    assert mock_create_bucket_err.call_count == 1  # Tries to create a bucket if at least one artifact needs to be uploaded
+    mock_create_bucket_err.assert_called_with(Bucket="default-us-east-1-1234")
+    assert spy_create_component.call_count == 0  # GG component is not created
```
Comment (pallavi03) [inline integration_tests/gdk/components/test_integ_PublishCommand.py:294]: if bucket exists , according to https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/main/gdk/commands/component/PublishCommand.py#L110 we should still create component right?

Comment (saranyailla) [reply integration_tests/gdk/components/test_integ_PublishCommand.py:294]: https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/main/gdk/commands/component/PublishCommand.py#L116

`BucketAlreadyExists` means that  another AWS account owns a bucket with the same name. We should exit asking for a different bucket here. It's different from `BucketAlreadyOwnedByYou` owned by you.

**Pull request merged**